source/FH8626V100/FH8626V100VideoCapturer.c

```
@@ -0,0 +1,292 @@
+
+    HANDLE_NULL_CHECK(handle);
+    HANDLE_GET(handle);
+
+    switch (resolution) {
+        case VID_RES_1080P:
+            chn = MAIN_CHN;
+            videoHandle->channelNum = chn;
+            res = FORMAT_1080P15;
+            break;
+        default:
+            kvs_log("Unsupported resolution %d", resolution);
+            return -EINVAL;
+    }
+
+    switch (format) {
+        case VID_FMT_H264:
+            fmt = FH_NORMAL_H264;
+            break;
+        default:
+            kvs_log("Unsupported format %d", format);
+            return -EINVAL;
+    }
+
+#ifdef USING_HARD_STREAM_VIDEO
+    if (change_video(chn, fmt, res)) {
+        kvs_log("change video failed\n");
+        return -EINVAL;
+    }
+#endif
+
+    videoHandle->format = format;
+    videoHandle->resolution = resolution;
+
+    return 0;
+}
+
+int videoCapturerGetFormat(const VideoCapturerHandle const handle, VideoFormat* pFormat, VideoResolution* pResolution)
+{
+    HANDLE_NULL_CHECK(handle);
+    HANDLE_GET(handle);
+
+    *pFormat = videoHandle->format;
+    *pResolution = videoHandle->resolution;
+
+    return 0;
+}
+
+int videoCapturerAcquireStream(VideoCapturerHandle handle)
+{
+    HANDLE_NULL_CHECK(handle);
+    HANDLE_GET(handle);
+
+    setStatus(handle, VID_CAP_STATUS_STREAM_ON);
+
+    return startRecvPic(handle, videoHandle->channelNum);
+}
+
+int videoCapturerGetFrame(VideoCapturerHandle handle, void* pFrameDataBuffer, const size_t frameDataBufferSize, uint64_t* pTimestamp,
+                          size_t* pFrameSize)
+{
+    int i, ret;
+    int frmlen, offset;
+    FH_VENC_STREAM stream;
+
+    HANDLE_NULL_CHECK(handle);
+    HANDLE_GET(handle);
+    HANDLE_STATUS_CHECK(videoHandle, VID_CAP_STATUS_STREAM_ON);
+
+    if (!pFrameDataBuffer || !pTimestamp || !pFrameSize) {
+        kvs_log("param err\n");
+        return -EINVAL;
+    }
+
+#ifdef USING_HARD_STREAM_VIDEO
+    if (videoHandle->format == VID_FMT_RAW) {
+        kvs_log("TODO VID_FMT_RAW");
```

If `RAW` format is not supported, `videoCapturerGetFrame` should return errno instead of print?